filestorage.go

```diff
@@ -82,12 +84,30 @@ func (s *FileStorage) Store(_ context.Context, key string, value []byte) error {
 	if err != nil {
 		return err
 	}
-	return os.WriteFile(filename, value, 0600)
+	fp, err := atomicfile.New(filename, 0o600)
+	if err != nil {
+		return err
+	}
+	_, err = fp.Write(value)
+	if err != nil {
+		// cancel the write
+		fp.Cancel()
```

Comment:
Why doesn't the fp automatically cancel the write when there's an error?

Reply:
cancel is a special call which deletes the temp file.

the fp to the tempfile can't know to delete itself when it closes.